### PR TITLE
Add RxJava3 support for androidx.paging.RxPagedListBuilder

### PR DESCRIPTION
**Pull request by poliver**
## Proposed Changes

Ports the rx2 `RxPagedListBuilder` to the rx3 package

## Testing

Test: `./gradlew test connectedCheck`

## Issues Fixed

Fixes: b/182497591

### CODE REVIEW

File: buildSrc/src/main/kotlin/androidx/build/LibraryVersions.kt [hunk 1]
```diff
@@ -90,7 +90,7 @@ object LibraryVersions {
     val MEDIAROUTER = Version("1.3.0-alpha01")
     val NAVIGATION = Version("2.4.0-alpha01")
     val NAVIGATION_COMPOSE = Version("1.0.0-alpha09")
-    val PAGING = Version("3.0.0-beta02")
+    val PAGING = Version("3.1.0-alpha01")
```
Comment (poliver) [inline buildSrc/src/main/kotlin/androidx/build/LibraryVersions.kt]: Not really sure what I was supposed to do here

Comment (dlam) [reply buildSrc/src/main/kotlin/androidx/build/LibraryVersions.kt]: Actually you can just revert this change - I'll handle the version bump in a separate PR and we won't be able to change this until the last beta is scheduled (which also means we'll have to wait to merge this PR).

**Review by dlam — COMMENTED** [review]
Proper format for the `Fixes:` tag is: `Fixes: 182497591`

File: buildSrc/src/main/kotlin/androidx/build/LibraryVersions.kt (hunk 1 above)
Comment (poliver) [reply buildSrc/src/main/kotlin/androidx/build/LibraryVersions.kt]: Is it sufficient to increment this locally for the purpose of running `./gradlew updateApi`, and just not commit it? `CONTRIBUTING.md` is a bit unclear.

Comment (dlam) [reply buildSrc/src/main/kotlin/androidx/build/LibraryVersions.kt]: We will unfortunately need to wait for next version bump to happen, and then rebase this PR and run `./gradlew updateApi` as we don't have a separate branch to track 3.1-alpha changes in parallel with 3.0-beta for reasons, but once paging goes rc, that'll happen on a release branch and I will re-ping here.

File: paging/rxjava3/src/androidTest/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  Copyright 2020 The Android Open Source Project
```
Comment (dlam) [inline paging/rxjava3/src/androidTest/AndroidManifest.xml]: 2021 here and elsewhere

**Comment by dlam** [comment]
Coming back to this now, once https://android-review.googlesource.com/c/platform/frameworks/support/+/1653527 lands we should be good to rebase this and get this landed :smiley: 

I'll update here once that happens as well. Sorry for the wait!

**Comment by dlam** [comment]
The version bump has landed now! Could you rebase this PR and re-run `./gradlew updateApi`?

**Comment by poliver** [comment]
Hey @dlam - I rebased this yesterday. Is there anything further needed here?

**Comment by dlam** [comment]
Nice - thanks for applying the new changes with `InitialPagedList()` too!

**Review by dlam — APPROVED** [review]

**Pull request closed without merging**